### DISCUSSION TITLE
Projection

### DISCUSSION THREAD

**Discussion by zjl114012** [maplibre/maplibre-native#1990] [Q&A]
The coordinates of the points I input are longitude and latitude. When I want to render them to the map through Opengl, I should need a projection. I am in the drawable_ Custom_ Layer_ Host_ Twicker.cpp found it
Mat4 projMatrix;
State. getProjMatrix (projMatrix);
Parameters. projectionMatrix=projMatrix;
Then input gl into the opengl tile shader_ Position=u_ Matrix * vec4 (a_pos, 0, 1); But this doesn't seem to be the projection transformation matrix I need either.

**Comment by louwers** [comment]
You don't need to drop down to the level of OpenGL to add points to the map. The custom drawable layer is not ready for public use yet.

Please explain a bit what you are trying to achieve, there is probably a better way to do it.
